sopel/irc/abstract_backends.py

```diff
@@ -0,0 +1,179 @@
+
+        :param list args: list of text, arguments of the IRC command to send
+        :param str text: optional text to send with the IRC command
+        :return: the raw message to send through the connection
+        :rtype: str
+
+        From :rfc:`2812` Internet Relay Chat: Client Protocol, Section 2.3:
+
+            IRC messages are always lines of characters terminated with a
+            CR-LF (Carriage Return - Line Feed) pair, and these messages SHALL
+            NOT exceed 512 characters in length, counting all characters
+            including the trailing CR-LF. Thus, there are 510 characters
+            maximum allowed for the command and its parameters. There is no
+            provision for continuation of message lines.
+
+        The length in the RFC refers to the length in *bytes*, which can be
+        bigger than the length of the unicode string. This method cuts the
+        message until its length fits within this limit of 510 bytes.
+
+        The returned message contains the CR-LF pair required at the end,
+        and can be sent as-is.
+        """
+        text = kwargs.get('text')
+        max_length = unicode_max_length = 510
+        raw_command = ' '.join(args)
+        if text is not None:
+            raw_command = '{args} :{text}'.format(args=raw_command,
+                                                  text=safe(text))
+
+        # The max length of 512 is in bytes, not unicode:
+        # we can't cut the message in bytes, or we may cut in the middle of a
+        # multi bytes characters.
+        while len(raw_command.encode('utf-8')) > max_length:
+            raw_command = raw_command[:unicode_max_length]
+            unicode_max_length = unicode_max_length - 1
+
+        # Ends the message with CR-LF
+        return raw_command + '\r\n'
+
+    def send_ping(self, host):
+        """Send a ``PING`` command to the server.
+
+        :param str host: IRC Server host
+
+        A ``PING`` command should be sent at a regular interval to make sure
+        the server knows the IRC connection is still active.
+        """
+        self.send_command('PING', safe(host))
+
+    def send_pong(self, host):
+        """Send a ``PONG`` command to the server.
+
+        :param str host: IRC Server host
+
+        A ``PONG`` command must be sent each time the server send a ``PING``
+        command to the client.
+        """
+        self.send_command('PONG', safe(host))
+
+    def send_nick(self, nick):
+        """Send a ``NICK`` command with a ``nick``.
+
+        :param str nick: Nick to take
+        """
+        self.send_command('NICK', safe(nick))
+
+    def send_user(self, user, mode, nick, name):
+        """Send a ``USER`` command with a ``user``.
+
+        :param str user: IRC username
+        :param str mode: List of mode to send for the user
+        :param str nick: Nick associated with this user
+        :param str name: Name for the user
```

Comment:
```suggestion
        :param str name: "real name" for the user
```

(That is what this field's for, right? I'm kinda lost in grammar-land rn. 😅)